ai-models/train_model.py

```
@@ -48,7 +48,44 @@ def __init__(self, data_path: str = "../data-pipeline/water_quality_data.db"):
+                """
+                # Use raw fetch to avoid Pandas sqlalchemy warning
+                cursor = conn.cursor()
+                cursor.execute(query)
+                columns = [desc[0] for desc in cursor.description]
+                data = cursor.fetchall()
+                df = pd.DataFrame(data, columns=columns)
+                cursor.close()
+                conn.close()
+                logger.info(f"Loaded {len(df)} records from PostgreSQL database")
+                return df
+            except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)